### ISSUE TITLE
Long title text overflow everywhere

### ISSUE THREAD

**Issue by tomsdev** [react-native-fellowship/react-native-navbar#124]
If you use a long title it doesn't truncate and instead overflow (because in your component style you use absolute positioning and doesn't set 'numberOfLines' on the Text component).
See: 
![screen shot 2015-12-21 at 5 09 28 pm](https://cloud.githubusercontent.com/assets/1284597/11935000/aa80d254-a805-11e5-9ab9-9e8f58ac0e98.png)

**Comment by grabbou** [comment]
Well, what's your expected behaviour? I would say that should be truncated on your side with e.g. `title.substring(0,10)` etc. if you are not sure about the length.

On the other hand - we could just set it up properly so there's native `...` added at the end and no multiline text is rendered.

**Comment by Kureev** [comment]
@grabbou I'll add `numberOfLines` to the `<Text />` element

**Comment by tomsdev** [comment]
Sorry I didn't set expectations, I think normally you want to have the text truncated with '...' At the end. It's not possible to use substring because you don't know the width available and the character size. I suggested to use numberOfLines and remove the absolute positioning.

**Comment by Kureev** [comment]
Sorry for the very late response, but can you compose a PR?

**Comment by grabbou** [comment]
@ex/navigator truncates, so I think we could also

**Comment by breadadams** [comment]
Any news on this? Running into the same issue with long titles

**Comment by ngankt2** [comment]
You can set maxHeight in style for Text
exp: <Text style={{maxHeight:22}}>blbalbalba very long over here</Text>

**Comment by breadadams** [comment]
Has there been an update for this @ngankt2? I'm sure I tried `numberOfLines={1}`, but the text still went behind back button.

**Comment by johncblandii** [comment]
Bump...

**Comment by chetstone** [comment]
I'm trying to use adjustsFontSizeToFit (RN >= 0.33) with a custom title component. It kind of works but the title is very short. It seems the default styles of this component provide equal space for the title and left and right buttons. I'd like the title to use more space. How can I specify that?

**Comment by Kureev** [comment]
IIRC, we using absolute positioning for the title, so no boundaries for the text, unless you specify them by yourself

**Comment by chetstone** [comment]
Yes. Ordinarily the title boundaries will expand to accommodate the
enclosed text, because the parent (styles.navBar) is flex with
justifyContent = space-between.

But when the space occupied by the text is dynamic, adjustsFontSizeToFit  is
trying to fit the text into a variable-size container and it ends up
deciding to use a very small font size. I'm not sure how to fix this.

**Comment by Kureev** [comment]
Yeah, it's a tricky one. I can refactor `position: absolute` to flexbox, but I'm not sure if it'll help anyhow. Wdyt?

**Comment by chetstone** [comment]
@kureev Thanks. I tried several changes but finally got it to work by adding `width` and `height` props to the `<Text>` component that has the `adjustsFontSizeToFit` prop. Without any changes to 'react-native-navbar`. 

Thanks
chester

**Comment by ngankt2** [comment]
@breadadams 
`<Text style={{**flex:1**}}` numberOfLines={1}>LONG TEXT....</Text>

**Comment by Kureev** [comment]
Thanks, @ngankt2.

Should we make it default for the title? Can somebody compose a PR?
